packages/core/src/config/index.ts

```diff
@@ -165,6 +165,17 @@ export async function resolveConfig(
     sortFarmJsPlugins
   );
 
+  if (config.compilation.mode === 'production' && !config.watch) {
```

Comment:
add a config option like `output.fileSize` and move this if statement to rust compiler similar to minify or tree shake.